RestoreVMFromRestorePoint.json

```diff
@@ -0,0 +1,249 @@
+    "contentVersion": "1.0.0.0",
+    "parameters": {
+        "location": {
+            "type": "string",
+            "defaultValue": "<location>"
+        },
+        "networkInterfaceName": {
+            "type": "string",
+            "defaultValue": "<nicName>"
+        },
+        "networkSecurityGroupName": {
+            "type": "string",
+            "defaultValue": "<nsgName>"
+        },
+        "subnetName": {
+            "type": "string",
+            "defaultValue": "<subnetName>"
+        },
+        "virtualNetworkName": {
+            "type": "string",
+            "defaultValue": "<vnetName>"
+        },
+        "publicIpAddressName": {
+            "type": "string",
+            "defaultValue": "<publicIPName>"
+        },
+        "virtualMachineName": {
+            "type": "string",
+            "defaultValue": "<vmName>"
+        },
+        "virtualMachineComputerName": {
+            "type": "string",
+            "defaultValue": "<computeName>"
+        },
+        "virtualMachineSize": {
+            "type": "string",
+            "defaultValue": "<vmSize>"
+        },
+        "adminUsername": {
+            "type": "string",
+            "defaultValue": "<username>"
+        },
+        "adminPassword": {
+            "type": "secureString",
+            "defaultValue": "<password>"
+        },
+        "osdiskRestorePointResourceId": {
+            "type": "string",
+            "defaultValue": "/subscriptions/<SubscriptionID>/resourceGroups/<ResourceGroupName>/providers/Microsoft.Compute/restorePointCollections/<RestorePointCollectionName>/restorePoints/<RestorePointName>/diskRestorePoints/<OSDiskRestorePointName>"
+        },
+        "datadiskRestorePointResourceId": {
+            "type": "string",
+            "defaultValue": "/subscriptions/<SubscriptionID>/resourceGroups/<ResourceGroupName>/providers/Microsoft.Compute/restorePointCollections/<RestorePointCollectionName>/restorePoints/<RestorePointName>/diskRestorePoints/<DataDiskRestorePointName>"
+        },
+        "restoredOSDiskName": {
+            "type": "string",
+            "defaultValue": "<restored_osdisk_name>"
+        },
+        "restoredOSDiskSku": {
+            "type": "string",
+            "defaultValue": "Premium_LRS"
+        },
+        "restoredOSDiskSizeGB": {
+            "type": "string",
+            "defaultValue": "1024"
+        },
+        "restoredDataDiskName": {
+            "type": "string",
+            "defaultValue": "<restored_datadisk_name>"
+        },
+        "restoredDataDiskSku": {
+            "type": "string",
+            "defaultValue": "Premium_LRS"
+        },
+        "restoredDataDiskSizeGB": {
+            "type": "string",
+            "defaultValue": "1024"
+        }
+    },
+    "variables": {
+        "nsgId": "[resourceId(resourceGroup().name, 'Microsoft.Network/networkSecurityGroups', parameters('networkSecurityGroupName'))]",
+        "vnetName": "[parameters('virtualNetworkName')]",
```

Comment:
Not required?

Reply:
Yes, not required. Doesn't break anything to have it either. ;)